core/translate/update.rs

```diff
@@ -183,6 +183,20 @@ pub fn prepare_update_plan(schema: &Schema, body: &mut Update) -> crate::Result<
         .map(|l| parse_limit(l))
         .unwrap_or(Ok((None, None)))?;
 
+    let mut indexes_to_update = vec![];
+    let indexes = schema.get_indices(&table_name.0);
+    for (set_column_index, _) in &set_clauses {
+        if let Some(index) = indexes.iter().find(|index| {
+            index
+                .columns
+                .iter()
+                .find(|column| column.pos_in_table == *set_column_index)
+                .is_some()
+        }) {
+            indexes_to_update.push(index.clone());
```

Comment:
Doesn't this add the same index multiple times potentially, or miss some indexes?

```sql
CREATE TABLE t(a);
CREATE INDEX a_asc on t(a);
CREATE INDEX a_desc on t(a desc);

UPDATE t SET a = 3;

// doesn't add a_desc because finds a_asc first
```

```sql
CREATE TABLE t(a,b,c);
CREATE INDEX i1 on t(a,b);
CREATE INDEX i2 on t(a,b desc);


UPDATE t SET a = 3, b=4;

// adds t1, then adds t1 again
```